test/libsolidity/SemanticTest.cpp

```diff
@@ -151,9 +151,18 @@ map<string, Builtin> SemanticTest::makeBuiltins() const
 			[this](FunctionCall const& _call) -> optional<bytes>
 			{
 				soltestAssert(_call.arguments.parameters.empty(), "No arguments expected.");
-				  return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
+				return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
 		 	}
-		}
+		},
+		{
+			"account",
+			[this](FunctionCall const& _call) -> optional<bytes>
+			{
+				soltestAssert(_call.arguments.parameters.size() == 1, "Account number expected.");
+				size_t accountNumber = static_cast<size_t>(stoi(_call.arguments.parameters.at(0).rawString));
+				return toBigEndian(u256(ExecutionFramework::setAccount(accountNumber).asBytes()));
```

Comment:
I don't think you need to get it into u256 again after `asBytes` as that serialises already:
```suggestion
				return ExecutionFramework::setAccount(accountNumber).asBytes();
```

I think this is also the reason of the test failure as the expectation turns into 32-bytes, but you have 20-bytes, i.e. needs the leading zeroes.

Reply:
So apparently `TestFunctionCall/BytesUtils` is rather broken and `BytesUtils::formatHex` will mess up the formatting. The proper solution (for now) is to right align this via:
```suggestion
				return toBigEndian(h256(ExecutionFramework::setAccount(accountNumber).asBytes(), h256::AlignRight));
```

At some point we may be able to fix this whole system not to have workarounds, but I'd say it is not worth it for the moment.